src/importer/sources/retroarch_source.py

```diff
@@ -0,0 +1,199 @@
+                boxart_folder_name = playlist_file.stem
+                image_path = (
+                    thumbnail_folder
+                    / boxart_folder_name
+                    / "Named_Boxarts"
+                    / boxart_image_name
+                )
+                additional_data = {"local_image_path": image_path}
+
+                yield (game, additional_data)
+
+        if bad_playlists:
+            raise FriendlyError(
+                _("No RetroArch Core Selected"),
+                # The variable is a newline separated list of playlists
+                _("The following playlists have no default core:")
+                + "\n\n{}\n\n".format("\n".join(bad_playlists))
+                + _("Games with no core selected were not imported"),
+            )
+
+
+class RetroarchLocations(NamedTuple):
+    config: Location
+
+
+class RetroarchSource(Source):
+    name = _("RetroArch")
+    source_id = "retroarch"
+    available_on = {"linux", "windows"}
+    iterable_class = RetroarchSourceIterable
+
+    locations = RetroarchLocations(
+        Location(
+            schema_key="retroarch-location",
+            candidates=[
+                shared.flatpak_dir / "org.libretro.RetroArch" / "config" / "retroarch",
+                shared.config_dir / "retroarch",
+                shared.home / ".config" / "retroarch",
+                Path("C:\\RetroArch-Win64"),
+                Path("C:\\RetroArch-Win32"),
+                shared.local_appdata_dir
+                / "Packages"
+                / "1e4cf179-f3c2-404f-b9f3-cb2070a5aad8_8ngdn9a6dx1ma"
+                / "LocalState",
+            ],
+            paths={
+                "retroarch.cfg": LocationSubPath("retroarch.cfg"),
+            },
+            invalid_subtitle=Location.CONFIG_INVALID_SUBTITLE,
+        )
+    )
+
+    @property
+    def executable_format(self):
+        self.locations.config.resolve()
+        is_flatpak = self.locations.config.root.is_relative_to(shared.flatpak_dir)
+        base = "flatpak run org.libretro.RetroArch" if is_flatpak else "retroarch"
+        args = '-L "{core_path}" "{rom_path}"'
+        return f"{base} {args}"
+
+    def __init__(self) -> None:
+        super().__init__()
+
+        try:
+            self.locations.config.candidates.append(self.get_steam_location())
+        except (OSError, KeyError, UnresolvableLocationError):
+            pass
+
+    def get_steam_location(self) -> str:
+        # Find steam location
+        libraryfolders = SteamSource().locations.data["libraryfolders.vdf"]
+        library_path = ""
+
+        with open(libraryfolders, "r", encoding="utf-8") as open_file:
+            for line in open_file:
+                if '"path"' in line:
+                    library_path = re.findall(
+                        '"path"\\s+"(.*)"\n', line, re.IGNORECASE
+                    )[0]
+                elif "1118310" in line:
```

Comment:
What does this elif do? It doesn't set the `library_path` variable, should it?
This would need a comment since it's not obvious.